src/App.tsx

```diff
@@ -1,27 +1,48 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+import 'bulma/css/bulma.css';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  // const [currentItems, setCurrentItems] = useState(items);
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const lastItemIndex = perPage * currentPage;
+  const firstItemIndex = lastItemIndex - perPage;
+  const preparedItems = items.slice(firstItemIndex, lastItemIndex);
+
+  const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setCurrentPage(1);
+    setPerPage(+e.target.value);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {firstItemIndex + 1} -{' '}
+        {lastItemIndex > items.length ? items.length : lastItemIndex} of{' '}
+        {items.length})
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
+            className="form-control"
+            onChange={handleChange}
+          >
             <option value="3">3</option>
-            <option value="5">5</option>
+            <option value="5" selected>
+              5
+            </option>
```

Comment:
Using the `selected` attribute on `<option>` is not recommended with React. Instead, set the `value` of the `<select>` element to match the desired option.